nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/util/PageProviderHelper.java

```diff
@@ -325,11 +325,14 @@ protected static String buildQueryStringWithPageProvider(PageProvider<?> provide
     @SuppressWarnings({ "rawtypes", "unchecked" })
     protected static String buildAggregatesClause(PageProvider provider) {
         try {
-            String aggregatesClause = "";
             // Aggregates that are being used as filters are stored in the namedParameters context data
             Properties namedParameters = (Properties) provider.getSearchDocumentModel()
                                                               .getContextData(NAMED_PARAMETERS);
             Map<String, Aggregate<? extends Bucket>> aggregates = provider.getAggregates();
+            if (namedParameters == null) {
+                return null;
```

Comment:
Any impact on returning `null` instead of `""` previously?
Ant the early exit could be placed one line before.

Reply:
So far nope (it is just used in org.nuxeo.ecm.automation.core.util.PageProviderHelper.buildQueryStringWithPageProvider(PageProvider, boolean) with a `StringUtils.isNotEmpty(aggregatesClause)`

But I agree we should keep it consistent.